ipn/ipnauth/access.go

```diff
@@ -0,0 +1,281 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package ipnauth
+
+import (
+	"math/bits"
+	"strconv"
+	"strings"
+)
+
+// DeviceAccess is a bitmask representing the requested, required, or granted
+// access rights to a device.
+type DeviceAccess uint32
+
+// ProfileAccess is a bitmask representing the requested, required, or granted
+// access rights to a Tailscale login profile.
+type ProfileAccess uint32
+
+const (
+	// ReadDeviceStatus is the access right required to read non-profile specific device statuses,
+	// such as the IP forwarding status. It is a non-privileged access right generally available to all users.
+	// It must not grant access to any sensitive or private information,
+	// including Tailscale profile names, network devices, etc.
+	ReadDeviceStatus DeviceAccess = 1 << iota
+	// GenerateBugReport is the access right required to generate a bug report
+	// (e.g. `tailscale bugreport` in CLI or Debug > Bug Report in GUI).
+	// It is a non-privileged access right granted to all users.
+	GenerateBugReport
+	// CreateProfile is the access right required to create new Tailscale profiles on the device.
+	// This operation is privileged on Unix-like platforms, including Linux,
+	// but is available to all users on non-Server Windows devices.
+	CreateProfile
+	// Debug is required for debugging operations that could expose sensitive information.
+	// Many such operations are accessible via `tailscale debug` subcommands.
+	// It is considered privileged access on all platforms, requiring root access on Unix-like systems
+	// and elevated admin access on Windows.
+	Debug
+	// InstallUpdates is required to initiate a Tailscale client self-update on platforms that support it.
+	// It is available to all users on all platforms except for Windows Server,
+	// where it requires admin rights.
+	InstallUpdates
+	// DeleteAllProfiles is required to log out from and delete all Tailscale profiles on a device.
+	// It is considered a privileged operation, requiring root access on Unix-like systems
+	// and elevated admin access on Windows.
+	DeleteAllProfiles
+
+	// UnrestrictedDeviceAccess combines all possible device access rights.
+	UnrestrictedDeviceAccess = ^DeviceAccess(0)
+)
+
+var deviceAccessNames = map[DeviceAccess]string{
+	CreateProfile:     "CreateProfile",
+	Debug:             "Debug",
+	DeleteAllProfiles: "DeleteAllProfiles",
+	GenerateBugReport: "GenerateBugReport",
+	InstallUpdates:    "InstallUpdates",
+	ReadDeviceStatus:  "ReadDeviceStatus",
+}
```

Comment:
cmd/stringer does this for you?

Reply:
`cmd/stringer` does not work with bit masks. But thanks for the suggestion, I'll keep it in mind!